framework/tlvf/src/include/tlvf/WSC/configData.h

```diff
@@ -60,7 +60,36 @@ class configData : public AttrList {
         auto attr = getAttr<cWscAttrMac>();
         return attr ? attr->data() : sMacAddr();
     };
-    uint8_t bss_type() const { return getAttr<cWscVendorExtWfa>()->subelement_value(); };
+    eWscVendorExtSubelementBssType bss_type() const
+    {
+        eWscVendorExtSubelementBssType default_value = eWscVendorExtSubelementBssType::TEARDOWN;
+
+        // Skip Vendor Extension attribute if it is not a WFA Vendor Extension attribute
+        auto vendor_ext_attr = getAttr<cWscAttrVendorExtension>();
+        if ((eWscVendorId::WSC_VENDOR_ID_WFA_1 != vendor_ext_attr->vendor_id_0()) ||
+            (eWscVendorId::WSC_VENDOR_ID_WFA_2 != vendor_ext_attr->vendor_id_1()) ||
+            (eWscVendorId::WSC_VENDOR_ID_WFA_3 != vendor_ext_attr->vendor_id_2())) {
+            return default_value;
+        }
+
+        size_t index     = 0;
+        auto vendor_data = vendor_ext_attr->vendor_data();
+        while ((index + sizeof(sWscWfaVendorExtSubelementMultiApIdentifier)) <=
+               vendor_ext_attr->vendor_data_length()) {
+            if (eWscWfaVendorExtSubelement::MULTI_AP_IDENTIFIER == vendor_data[index++]) {
+                if (0x01 != vendor_data[index++]) {
+                    return default_value;
+                }
+
+                return static_cast<WSC::eWscVendorExtSubelementBssType>(vendor_data[index]);
+            }
+
+            // Skip this subelement
+            index += vendor_data[index++];
```

Comment:
So this is why I don't like `index++`... It *looks* like index is incremented 3 times at first sight, you have to really follow the flow to understand that this is `vendor_data[index + 1]`...
```suggestion
            index += 2 + subelement_length;
```

Reply:
Done